packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -77,7 +80,9 @@ function HeaderToolbar( { onToggleInserter, isInserterOpen } ) {
 			<EditorHistoryUndo />
 			<EditorHistoryRedo />
 			<TableOfContents hasOutlineItemsDisabled={ isTextModeEnabled } />
-			<BlockNavigationDropdown isDisabled={ isTextModeEnabled } />
+			<BlockNavigationContext.Provider value={ blockNavigationContext }>
+				<BlockNavigationDropdown isDisabled={ isTextModeEnabled } />
+			</BlockNavigationContext.Provider>
```

Comment:
I think the provider could be useful, as there are a few places where props are passed through multiple layers internally in the block navigation components (though some of that is not very visible due to the recursive nature of the component.)

It feels like the `BlockNavigationContext.Provider` should be an internal detail though. Probably as part of a wrapper around the top level `BlockNavigationList`.

Reply:
If the provider was an internal detail, only used in `block-navigation/index.js`, how would you control the value of `withBlockNavigationSlots` from the `HeaderToolbar`?

Reply:
An option would be to pass it as a prop to `BlockNavigationDropdown` and then pass that prop to the context provider. I know that defeats some of the purpose of context, but it still achieves the goal of using context to pass the setting down to the leaf navigation item.